cg/services/slurm_upload_service/slurm_upload_service.py

```diff
@@ -0,0 +1,43 @@
+from cg.apps.tb.api import TrailblazerAPI
+from cg.apps.tb.models import TrailblazerAnalysis
+from cg.constants.priority import SlurmQos
+from cg.constants.slurm import (
+    SLURM_UPLOAD_EXCLUDED_COMPUTE_NODES,
+    SLURM_UPLOAD_MAX_HOURS,
+    SLURM_UPLOAD_MEMORY,
+)
+from cg.models.slurm.sbatch import Sbatch
+from cg.services.slurm_service.slurm_service import SlurmService
+from cg.services.slurm_upload_service.slurm_upload_config import SlurmUploadConfig
+from cg.services.slurm_upload_service.utils import get_quality_of_service
+
+
+class SlurmUploadService:
+    def __init__(
+        self,
+        slurm_service: SlurmService,
+        trailblazer_api: TrailblazerAPI,
+        config: SlurmUploadConfig,
+    ):
+        self.slurm_service = slurm_service
+        self.trailblazer_api = trailblazer_api
+        self.config = config
+
+    def upload(self, upload_command: str, job_name: str, case_id: str):
+        analysis: TrailblazerAnalysis = self.trailblazer_api.get_latest_analysis(case_id)
```

Comment:
If it is not too much trouble I would like this to be latest_completed analysis.

Reply:
That would mean changing the logic in the endpoint in trailblazer, alternatively adding a new endpoint for it. Do we always want the most recent completed one?

Reply:
Yes, we never want to add it to one that is not completed. I'm scratching my head regarding how likely it is to have a completed analysis and then a newer running one. I think it is reasonable likely to occur a couple of times per month. I think it makes sense for the trailblazer endpoint to be able to filter on status.

Reply:
Added some filters to the endpoint in trailblazer and retrieve the analysis here using them